haas/api.py

```diff
@@ -612,7 +708,8 @@ def port_detach_nic(port):
     """
     db = model.Session()
 
-    port = _must_find(db, model.Port, port)
+    switch = _must_find(db, model.Switch, switch)
+    port = _must_find_n(db, switch, model.Port, port)
 
```

Comment:
Actually, there are other constraints: what the nic is attached to networks, for example? I think the thing to check is that the node is free; that will ensure everything else is already detached.